Sentiment Analysis: A study of Lexical Method, Neural Networks and SVMs
=====================================================================

In this project, we try to estimate the sentiment associated with reviews from Amazon, Yelp, and IMDB and how they perform on different review types (restaurant, movies etc). Our first model is a lexicon based sentiment analyzer
----

In [3]:
# Making the necessary imports
import numpy as np
import pandas as pd

# NLP toolkit
import nltk
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer
from nltk.stem import SnowballStemmer
from nltk import word_tokenize
from nltk.tokenize import sent_tokenize

# Regex in heuristics
import re

# For NN
from sklearn.neural_network import MLPClassifier
# For SVM
from sklearn import svm

In [4]:
# nltk stopwords
stop = stopwords.words('english')

In [5]:
print stop

[u'i', u'me', u'my', u'myself', u'we', u'our', u'ours', u'ourselves', u'you', u'your', u'yours', u'yourself', u'yourselves', u'he', u'him', u'his', u'himself', u'she', u'her', u'hers', u'herself', u'it', u'its', u'itself', u'they', u'them', u'their', u'theirs', u'themselves', u'what', u'which', u'who', u'whom', u'this', u'that', u'these', u'those', u'am', u'is', u'are', u'was', u'were', u'be', u'been', u'being', u'have', u'has', u'had', u'having', u'do', u'does', u'did', u'doing', u'a', u'an', u'the', u'and', u'but', u'if', u'or', u'because', u'as', u'until', u'while', u'of', u'at', u'by', u'for', u'with', u'about', u'against', u'between', u'into', u'through', u'during', u'before', u'after', u'above', u'below', u'to', u'from', u'up', u'down', u'in', u'out', u'on', u'off', u'over', u'under', u'again', u'further', u'then', u'once', u'here', u'there', u'when', u'where', u'why', u'how', u'all', u'any', u'both', u'each', u'few', u'more', u'most', u'other', u'some', u'such', u'no', u'nor', u

These stopwords contain important words like 'not' and "n't" that are informative in reviews. So we edit these stopwords.
---

In [6]:
stop_edit = [u'i', u'me', u'my', u'myself', u'we', u'our', u'ours', u'ourselves', u'you', u'your', u'yours', u'yourself', u'yourselves', u'he', u'him', u'his', u'himself', u'she', u'her', u'hers', u'herself', u'it', u'its', u'itself', u'they', u'them', u'their', u'theirs', u'themselves', u'what', u'which', u'who', u'whom', u'this', u'that', u'these', u'those', u'am', u'is', u'are', u'was', u'were', u'be', u'been', u'being', u'have', u'has', u'had', u'having', u'do', u'does', u'did', u'doing', u'a', u'an', u'the', u'and', u'but', u'if', u'or', u'because', u'as', u'until', u'while', u'of', u'at', u'by', u'for', u'with', u'about', u'between', u'into', u'through', u'during', u'before', u'after', u'above', u'below', u'to', u'from', u'up', u'down', u'in', u'out', u'on', u'off', u'over', u'under', u'again', u'further', u'then', u'once', u'here', u'there', u'when', u'where', u'why', u'how', u'all', u'any', u'both', u'each', u'few', u'more', u'most', u'other', u'some', u'such', u'only', u'own', u'same', u'so', u'than', u'too', u'very', u'can', u'will', u'just', u'should', u'now']

In [7]:
stop_edit

[u'i',
 u'me',
 u'my',
 u'myself',
 u'we',
 u'our',
 u'ours',
 u'ourselves',
 u'you',
 u'your',
 u'yours',
 u'yourself',
 u'yourselves',
 u'he',
 u'him',
 u'his',
 u'himself',
 u'she',
 u'her',
 u'hers',
 u'herself',
 u'it',
 u'its',
 u'itself',
 u'they',
 u'them',
 u'their',
 u'theirs',
 u'themselves',
 u'what',
 u'which',
 u'who',
 u'whom',
 u'this',
 u'that',
 u'these',
 u'those',
 u'am',
 u'is',
 u'are',
 u'was',
 u'were',
 u'be',
 u'been',
 u'being',
 u'have',
 u'has',
 u'had',
 u'having',
 u'do',
 u'does',
 u'did',
 u'doing',
 u'a',
 u'an',
 u'the',
 u'and',
 u'but',
 u'if',
 u'or',
 u'because',
 u'as',
 u'until',
 u'while',
 u'of',
 u'at',
 u'by',
 u'for',
 u'with',
 u'about',
 u'between',
 u'into',
 u'through',
 u'during',
 u'before',
 u'after',
 u'above',
 u'below',
 u'to',
 u'from',
 u'up',
 u'down',
 u'in',
 u'out',
 u'on',
 u'off',
 u'over',
 u'under',
 u'again',
 u'further',
 u'then',
 u'once',
 u'here',
 u'there',
 u'when',
 u'where',
 u'why',
 u'how',
 u'all',
 u'any',
 

In [8]:
# Loads the reviews and returns training, test and validation sets.
def load_reviews():
    df = pd.read_csv('data.txt', delimiter='\t', header=None)
    df.columns = ['Text', 'Sentiment']
    return np.split(df.sample(frac=1), [int(.6*len(df)), int(.8*len(df))])

In [9]:
train, validate, test = load_reviews()

In [10]:
# Combine train and validate for the lexicon based method because there are no parameters to tune.
train_validate = train.append(validate)

In [11]:
def ltz(s):
    """
    Function that lemmatizes a given token
    """
    lemmatizer = WordNetLemmatizer()
    a = lemmatizer.lemmatize(s, pos='n')
    b = lemmatizer.lemmatize(s, pos='v')
    if len(a) <= len(b):
        return a
    return b

In [12]:
def compress_word(s):
    """
    Function that compresses stressed words like 'coooooooooooooollllllllllllll' and 'wwwwwwwwaaaaaaaaaayyyyyyy'
    """
    patterns = [r'\bw+a+y+\b', r'\bc+o+o+l+\b', r'\bn+i+c+e+\b']
    matches = ['way', 'cool', 'nice']
    for i in range(len(patterns)):
        p = patterns[i]
        m = matches[i]
        rx = re.compile(p)
        if re.match(p, s):
            return matches[i]
    return s

In [13]:
compress_word('coooooooooooooooooooooooolllllllllll')

'cool'

In [14]:
def preprocessor(text):
    """
    Function that preprocesses the input string, based on some heuristics.
    """
    regex = re.compile('\.{2,}')
    regex2 = re.compile(r'(\w+)\.(\w+)')
    sentences = sent_tokenize(text)
    total = []
    for s in sentences:
        txt = regex2.sub(r'\1. \2', s)
        tokens = word_tokenize(txt)
        tokens = [s.lower() for s in tokens]
        tokens = [unicode(x, errors='ignore') for x in tokens]
        tokens = [regex.sub('', x) for x in tokens]
        tokens = [x for x in tokens if x not in '.,;!?']
        tokens = [x for x in tokens if len(x)!=0]
        tokens = [x if x != "n't" else "not" for x in tokens]
        tokens = [x if x != "'s" else "is" for x in tokens]
        tokens = [x if x != "'d" else "would" for x in tokens]
        tokens = map(ltz, tokens)
        tokens = [x for x in tokens if x not in stop_edit]
        tokens = map(compress_word, tokens)
        total = total + tokens
        total.append("<TERM>")
    return total

In [15]:
# Batch preprocessor
def batch_preprocessor(inp):
    data_ = []
    # Assuming that the columns are 'Text' and 'Sentiment'
    for index, row in inp.iterrows():
        print row['Text']
        total = preprocessor(row['Text'])
        print total
        total = ' '.join(total)
        print total
        print '==========================================================================================='
        data_.append((total, row['Sentiment']))
    return pd.DataFrame(data_)

In [16]:
curated = batch_preprocessor(train_validate)

I cannot make calls at certain places.
[u'not', u'make', u'call', u'certain', u'place', '<TERM>']
not make call certain place <TERM>
His co-stars don't fare much better, with people like Morgan Freeman, Jonah Hill, and Ed Helms just wasted.  
[u'co-star', 'not', u'fare', u'much', u'better', u'people', u'like', u'morgan', u'freeman', u'jonah', u'hill', u'ed', u'helm', u'waste', '<TERM>']
co-star not fare much better people like morgan freeman jonah hill ed helm waste <TERM>
The guy who said he's had better dialogue with his potted plants has it right.  
[u'guy', u'say', u'better', u'dialogue', u'pot', u'plant', u'ha', u'right', '<TERM>']
guy say better dialogue pot plant ha right <TERM>
10/10  
[u'10/10', '<TERM>']
10/10 <TERM>
This is a great restaurant at the Mandalay Bay.
[u'great', u'restaurant', u'mandalay', u'bay', '<TERM>']
great restaurant mandalay bay <TERM>
The "1.2" mega pixel camera, being a part of a phone, is reasonably good7.
[u'``', u'1', u'2', u"''", u'mega', u'pixel', 

A great touch.
[u'great', u'touch', '<TERM>']
great touch <TERM>
Great Earpiece.
[u'great', u'earpiece', '<TERM>']
great earpiece <TERM>
If there was ever an indication of a writer and a director's ability to meld two highly volatile temperaments into a seamless union of creativity, then this is it!  
[u'wa', u'ever', u'indication', u'writer', u'director', u'ability', u'meld', u'two', u'highly', u'volatile', u'temperament', u'seamless', u'union', u'creativity', '<TERM>']
wa ever indication writer director ability meld two highly volatile temperament seamless union creativity <TERM>
The original Body and Soul (1947) is a masterpiece.  
[u'original', u'body', u'soul', u'(', u'1947', u')', u'masterpiece', '<TERM>']
original body soul ( 1947 ) masterpiece <TERM>
The pleather case doesn't fit.
[u'pleather', u'case', 'not', u'fit', '<TERM>']
pleather case not fit <TERM>
My father has the V265, and the battery is dying.
[u'father', u'ha', u'v265', u'battery', u'die', '<TERM>']
father ha v265 

The range is very decent, I've been able to roam around my house with the phone in the living room with no reception/sound quality issues.
[u'range', u'decent', u"'ve", u'able', u'roam', u'around', u'house', u'phone', u'live', u'room', u'no', u'reception/sound', u'quality', u'issue', '<TERM>']
range decent 've able roam around house phone live room no reception/sound quality issue <TERM>
John Garfield, Ann revere, Lilli Plmer, William Conrad, Canada Lee...and filmed by one of the greatest cinematographers to ever grace the screen..James Wong Howe.  
[u'john', u'garfield', u'ann', u'revere', u'lilli', u'plmer', u'william', u'conrad', u'canada', u'lee', u'film', u'one', u'greatest', u'cinematographer', u'ever', u'grace', u'screenjames', u'wong', u'howe', '<TERM>']
john garfield ann revere lilli plmer william conrad canada lee film one greatest cinematographer ever grace screenjames wong howe <TERM>
It's one of the most boring Horror movies from the 90's mainly because it starts slow and 

[u'really', u'enjoy', u'eat', '<TERM>']
really enjoy eat <TERM>
Bad characters, bad story and bad acting.  
[u'bad', u'character', u'bad', u'story', u'bad', u'act', '<TERM>']
bad character bad story bad act <TERM>
Kathy Bates is wonderful in her characters subtle desperation and escapism; a variation on her character in "At Play In The Fields Of The Lord".  
[u'kathy', u'bat', u'wonderful', u'character', u'subtle', u'desperation', u'escapism', u'variation', u'character', u'``', u'play', u'field', u'lord', u"''", '<TERM>']
kathy bat wonderful character subtle desperation escapism variation character `` play field lord '' <TERM>
This is essentially a communications tool that does not communicate.
[u'essentially', u'communication', u'tool', u'not', u'communicate', '<TERM>']
essentially communication tool not communicate <TERM>
Very unreliable service from T-mobile !
[u'unreliable', u'service', u't-mobile', '<TERM>']
unreliable service t-mobile <TERM>
But the duet between the astronaut and

I love this thing!
[u'love', u'thing', '<TERM>']
love thing <TERM>
It's very convenient and simple to use - gets job done & makes the car ride so much smoother.
[u'convenient', u'simple', u'use', u'-', u'get', u'job', u'&', u'make', u'car', u'ride', u'much', u'smoother', '<TERM>']
convenient simple use - get job & make car ride much smoother <TERM>
All in all a beautiful directed film from Nicola's roeg wih a sublime cast.  
[u'beautiful', u'direct', u'film', u'nicola', u'roeg', u'wih', u'sublime', u'cast', '<TERM>']
beautiful direct film nicola roeg wih sublime cast <TERM>
One of my favorite purchases ever.
[u'one', u'favorite', u'purchase', u'ever', '<TERM>']
one favorite purchase ever <TERM>
I had heard good things about this place, but it exceeding every hope I could have dreamed of.
[u'hear', u'good', u'thing', u'place', u'exceed', u'every', u'hope', u'could', u'dream', '<TERM>']
hear good thing place exceed every hope could dream <TERM>
GO AND SEE IT!  
[u'go', u'see', '<TERM>']


everything act cinematography wa solid <TERM>
I wasn't really impressed with Strip Steak.
[u'wa', 'not', u'really', u'impress', u'strip', u'steak', '<TERM>']
wa not really impress strip steak <TERM>
Highly entertaining at all angles.  
[u'highly', u'entertain', u'angle', '<TERM>']
highly entertain angle <TERM>
If she had not rolled the eyes we may have stayed... Not sure if we will go back and try it again.
[u'not', u'roll', u'eye', u'may', u'stay', u'not', u'sure', u'go', u'back', u'try', '<TERM>']
not roll eye may stay not sure go back try <TERM>
Watching washing machine twirling around wouldn't hurt your eyes as much as this show.  
[u'watch', u'wash', u'machine', u'twirl', u'around', u'would', 'not', u'hurt', u'eye', u'much', u'show', '<TERM>']
watch wash machine twirl around would not hurt eye much show <TERM>
How awful she is!  
[u'awful', '<TERM>']
awful <TERM>
Headset works great & was packaged nicely to avoid any damage.
[u'headset', u'work', u'great', u'&', u'wa', u'package',

But it's just not funny.  
[u'not', u'funny', '<TERM>']
not funny <TERM>
It's one of the movies I need to see whenever it comes on TV...never mind the fact that I already have it memorized!  
[u'one', u'movie', u'need', u'see', u'whenever', u'come', u'tv', u'never', u'mind', u'fact', u'already', u'memorize', '<TERM>']
one movie need see whenever come tv never mind fact already memorize <TERM>
IMDB ratings only go as low 1 for awful, it's time to get some negative numbers in there for cases such as these.  
[u'imdb', u'rat', u'go', u'low', u'1', u'awful', u'time', u'get', u'negative', u'number', u'case', '<TERM>']
imdb rat go low 1 awful time get negative number case <TERM>
I guess I should have known that this place would suck, because it is inside of the Excalibur, but I didn't use my common sense.
[u'guess', u'know', u'place', u'would', u'suck', u'inside', u'excalibur', 'not', u'use', u'common', u'sense', '<TERM>']
guess know place would suck inside excalibur not use common sense <TE

It was extremely "crumby" and pretty tasteless.
[u'wa', u'extremely', u'``', u'crumby', u"''", u'pretty', u'tasteless', '<TERM>']
wa extremely `` crumby '' pretty tasteless <TERM>
The chicken was deliciously seasoned and had the perfect fry on the outside and moist chicken on the inside.
[u'chicken', u'wa', u'deliciously', u'season', u'perfect', u'fry', u'outside', u'moist', u'chicken', u'inside', '<TERM>']
chicken wa deliciously season perfect fry outside moist chicken inside <TERM>
Product was excellent and works better than the verizon one and Boy was it cheaper!
[u'product', u'wa', u'excellent', u'work', u'better', u'verizon', u'one', u'boy', u'wa', u'cheaper', '<TERM>']
product wa excellent work better verizon one boy wa cheaper <TERM>
How this piece of trash was ever released is beyond me: the acting, the story, the characters, the supposedly special effects, etc...it's ALL wrong.  
[u'piece', u'trash', u'wa', u'ever', u'release', u'beyond', u':', u'act', u'story', u'character', 

I don't like this Nokia either.
['not', u'like', u'nokia', u'either', '<TERM>']
not like nokia either <TERM>
You won't forget this movie!  
[u'wo', 'not', u'forget', u'movie', '<TERM>']
wo not forget movie <TERM>
They could serve it with just the vinaigrette and it may make for a better overall dish, but it was still very good.
[u'could', u'serve', u'vinaigrette', u'may', u'make', u'better', u'overall', u'dish', u'wa', u'still', u'good', '<TERM>']
could serve vinaigrette may make better overall dish wa still good <TERM>
Display is excellent and camera is as good as any from that year.
[u'display', u'excellent', u'camera', u'good', u'year', '<TERM>']
display excellent camera good year <TERM>
Also great directing and photography.  
[u'also', u'great', u'direct', u'photography', '<TERM>']
also great direct photography <TERM>
I am so thrilled after seeing a movie like this.  
[u'thrill', u'see', u'movie', u'like', '<TERM>']
thrill see movie like <TERM>
A must study for anyone interested in

One nice thing was that they added gratuity on the bill since our party was larger than 6 or 8, and they didn't expect more tip than that.
[u'one', 'nice', u'thing', u'wa', u'add', u'gratuity', u'bill', u'since', u'party', u'wa', u'larger', u'6', u'8', 'not', u'expect', u'tip', '<TERM>']
one nice thing wa add gratuity bill since party wa larger 6 8 not expect tip <TERM>
Not too screamy not to masculine but just right.  
[u'not', u'screamy', u'not', u'masculine', u'right', '<TERM>']
not screamy not masculine right <TERM>
I was not happy with this item.
[u'wa', u'not', u'happy', u'item', '<TERM>']
wa not happy item <TERM>
It doesn't make you look cool.
['not', u'make', u'look', 'cool', '<TERM>']
not make look cool <TERM>
The acting is fantastic, the stories are seamlessly woven together, and the dogs are splendid............. A must rent, view, or see.  
[u'act', u'fantastic', u'story', u'seamlessly', u'woven', u'together', u'dog', u'splendid', u'must', u'rent', u'view', u'see', '<TERM>'

Highly unrecommended.  
[u'highly', u'unrecommended', '<TERM>']
highly unrecommended <TERM>
(My mother and brother had to do this)When I saw Robert Ryan portraying this type of man, it was a very good imitation of this type of individual, and I was impressed.  
[u'(', u'mother', u'brother', u')', u'saw', u'robert', u'ryan', u'portray', u'type', u'man', u'wa', u'good', u'imitation', u'type', u'individual', u'wa', u'impress', '<TERM>']
( mother brother ) saw robert ryan portray type man wa good imitation type individual wa impress <TERM>
Service is quick and even "to go" orders are just like we like it!
[u'service', u'quick', u'even', u'``', u'go', u"''", u'order', u'like', u'like', '<TERM>']
service quick even `` go '' order like like <TERM>
I had a seriously solid breakfast here.
[u'seriously', u'solid', u'breakfast', '<TERM>']
seriously solid breakfast <TERM>
The Jamaican mojitos are delicious.
[u'jamaican', u'mojitos', u'delicious', '<TERM>']
jamaican mojitos delicious <TERM>
It's as

[u'someone', u'strive', u'greatness', u'poetry', u'deliver', u'muddle', u'(', u'often', u'ridiculous', u')', u'story', u'bunch', u'disparate', u'scene', u'pretentious', u'dialogue', u'get', u'worst', u'kind', u'movie', u'reviewer', u'accurately', u'define', u'``', u'pretentious', u'crap', u"''", '<TERM>']
someone strive greatness poetry deliver muddle ( often ridiculous ) story bunch disparate scene pretentious dialogue get worst kind movie reviewer accurately define `` pretentious crap '' <TERM>
I love this phone!.
[u'love', u'phone', '<TERM>']
love phone <TERM>
The last 3 times I had lunch here has been bad.
[u'last', u'3', u'time', u'lunch', u'ha', u'bad', '<TERM>']
last 3 time lunch ha bad <TERM>
It presents a idyllic yet serious portrayal of the ups and downs of the characters lives.  
[u'present', u'idyllic', u'yet', u'serious', u'portrayal', u'character', u'life', '<TERM>']
present idyllic yet serious portrayal character life <TERM>
The best electronics of the available FM Trans

Really pleased with this product so far.
[u'really', u'please', u'product', u'far', '<TERM>']
really please product far <TERM>
I have purchased these for both family and friends, and all enjoy their clarity and ease of use
[u'purchase', u'family', u'friend', u'enjoy', u'clarity', u'ease', u'use', '<TERM>']
purchase family friend enjoy clarity ease use <TERM>
Poorly contstruct hinge.
[u'poorly', u'contstruct', u'hinge', '<TERM>']
poorly contstruct hinge <TERM>
Damian is so talented and versatile in so many ways of writing and portraying different Characters on screen.  
[u'damian', u'talented', u'versatile', u'many', 'way', u'write', u'portray', u'different', u'character', u'screen', '<TERM>']
damian talented versatile many way write portray different character screen <TERM>
We had fantastic service, and were pleased by the atmosphere.
[u'fantastic', u'service', u'please', u'atmosphere', '<TERM>']
fantastic service please atmosphere <TERM>
An Awesome New Look For Fall 2000!.
[u'awesome'

[u'another', u'note', u'phone', u'appearance', u'really', u'look', u'rather', u'bland', u'especially', u'black', u'model', '<TERM>']
another note phone appearance really look rather bland especially black model <TERM>
The movie seemed a little slow at first.  
[u'movie', u'seem', u'little', u'slow', u'first', '<TERM>']
movie seem little slow first <TERM>
The majority of the Logitech earbud headsets failed.
[u'majority', u'logitech', u'earbud', u'headset', u'fail', '<TERM>']
majority logitech earbud headset fail <TERM>
They could be used as exemplars for any set designer.  
[u'could', u'use', u'exemplar', u'set', u'designer', '<TERM>']
could use exemplar set designer <TERM>
Aside from it's terrible lead, this film has loads of other debits.  
[u'aside', u'terrible', u'lead', u'film', u'ha', u'load', u'debit', '<TERM>']
aside terrible lead film ha load debit <TERM>
I was so insulted.
[u'wa', u'insult', '<TERM>']
wa insult <TERM>
Worst ever.
[u'worst', u'ever', '<TERM>']
worst ever <TERM>

[u'put', u'latest', u'o', u'(', u'v1', u'15g', u')', u'like', u'slow', u'crawl', u'lock', u'every', '<TERM>']
put latest o ( v1 15g ) like slow crawl lock every <TERM>
The bus boy on the other hand was so rude.
[u'bus', u'boy', u'hand', u'wa', u'rude', '<TERM>']
bus boy hand wa rude <TERM>
The food was outstanding and the prices were very reasonable.
[u'food', u'wa', u'outstanding', u'price', u'reasonable', '<TERM>']
food wa outstanding price reasonable <TERM>
I love this place.
[u'love', u'place', '<TERM>']
love place <TERM>
Barney is definitely super DE duper.  
[u'barney', u'definitely', u'super', u'de', u'duper', '<TERM>']
barney definitely super de duper <TERM>
Paolo Sorrentino has written a wonderful story about loneliness and Tony has built one of the most unforgettable characters seen in movies in recent years.  
[u'paolo', u'sorrentino', u'ha', u'write', u'wonderful', u'story', u'loneliness', u'tony', u'ha', u'built', u'one', u'unforgettable', u'character', u'see', u'movie', u

This is so embarassing and also my ears hurt if I try to push the ear plug into my ear.
[u'embarassing', u'also', u'ear', u'hurt', u'try', u'push', u'ear', u'plug', u'ear', '<TERM>']
embarassing also ear hurt try push ear plug ear <TERM>
Plus, it was well-paced and suited its relatively short run time.  
[u'plus', u'wa', u'well-paced', u'suit', u'relatively', u'short', u'run', u'time', '<TERM>']
plus wa well-paced suit relatively short run time <TERM>
It fits comfortably in either ear, the sound is clear and loud, and the charge lasts a couple of days.
[u'fit', u'comfortably', u'either', u'ear', u'sound', u'clear', u'loud', u'charge', u'last', u'couple', u'day', '<TERM>']
fit comfortably either ear sound clear loud charge last couple day <TERM>
I contacted the company and they told me that, although the unit was still under warranty, if I couldn't produce my receipt I was out of luck.
[u'contact', u'company', u'told', u'although', u'unit', u'wa', u'still', u'warranty', u'could', 'not',

[u'steer', u'clear', u'product', u'go', u'genuine', u'palm', u'replacementr', u'pen', u'come', u'three-pack', '<TERM>']
steer clear product go genuine palm replacementr pen come three-pack <TERM>
Great Pizza and Salads!
[u'great', u'pizza', u'salad', '<TERM>']
great pizza salad <TERM>
It's hard not to fall head-over-heels in love with that girl.  
[u'hard', u'not', u'fall', u'head-over-heels', u'love', u'girl', '<TERM>']
hard not fall head-over-heels love girl <TERM>
The only reason to eat here would be to fill up before a night of binge drinking just to get some carbs in your stomach.
[u'reason', u'eat', u'would', u'fill', u'night', u'binge', u'drink', u'get', u'carbs', u'stomach', '<TERM>']
reason eat would fill night binge drink get carbs stomach <TERM>
It was a riot to see Hugo Weaving play a sex-obsessed gay real estate salesman who uses his clients' houses for his trysts with the flaming Darren (Tom Hollander).  
[u'wa', u'riot', u'see', u'hugo', u'weave', u'play', u'sex-obsessed

-My order was not correct.
[u'-my', u'order', u'wa', u'not', u'correct', '<TERM>']
-my order wa not correct <TERM>
They also now serve Indian naan bread with hummus and some spicy pine nut sauce that was out of this world.
[u'also', u'serve', u'indian', u'naan', u'bread', u'hummus', u'spicy', u'pine', u'nut', u'sauce', u'wa', u'world', '<TERM>']
also serve indian naan bread hummus spicy pine nut sauce wa world <TERM>
I could not stand to even watch it for very long for fear of losing I.Q.  
[u'could', u'not', u'stand', u'even', u'watch', u'long', u'fear', u'lose', u'i.', u'q', '<TERM>']
could not stand even watch long fear lose i. q <TERM>
Didn't laugh or even smile once.  
['not', u'laugh', u'even', u'smile', '<TERM>']
not laugh even smile <TERM>
I bought this to use with my Kindle Fire and absolutely loved it!
[u'buy', u'use', u'kindle', u'fire', u'absolutely', u'love', '<TERM>']
buy use kindle fire absolutely love <TERM>
To call this movie a drama is ridiculous!  
[u'call', u'movie'

This particular model would not work with my Motorola Q smartphone.
[u'particular', u'model', u'would', u'not', u'work', u'motorola', u'q', u'smartphone', '<TERM>']
particular model would not work motorola q smartphone <TERM>
The service was outshining & I definitely recommend the Halibut.
[u'service', u'wa', u'outshine', u'&', u'definitely', u'recommend', u'halibut', '<TERM>']
service wa outshine & definitely recommend halibut <TERM>
Don't miss it.  
['not', u'miss', '<TERM>']
not miss <TERM>
I Was Hoping for More.
[u'wa', u'hop', '<TERM>']
wa hop <TERM>
AMAZON SUCKS.
[u'amazon', u'suck', '<TERM>']
amazon suck <TERM>
Very Dissapointing Performance.
[u'dissapointing', u'performance', '<TERM>']
dissapointing performance <TERM>
No buyers remorse on this one!.
[u'no', u'buyer', u'remorse', u'one', '<TERM>']
no buyer remorse one <TERM>
The food was very good and I enjoyed every mouthful, an enjoyable relaxed venue for couples small family groups etc.
[u'food', u'wa', u'good', u'enjoy', u'e

[u'fit', u'securely', u'ear', u'hook', u'not', u'even', u'need', u'use', u'sound', u'better', u'direct', u'ear', u'canal', '<TERM>']
fit securely ear hook not even need use sound better direct ear canal <TERM>
I received my orders well within the shipping timeframe, everything was in good working order and overall, I am very excited to have this source.
[u'receive', u'order', u'well', u'within', u'ship', u'timeframe', u'everything', u'wa', u'good', u'work', u'order', u'overall', u'excite', u'source', '<TERM>']
receive order well within ship timeframe everything wa good work order overall excite source <TERM>
Absolutely great.
[u'absolutely', u'great', '<TERM>']
absolutely great <TERM>
Perfect for someone (me) who only likes beer ice cold, or in this case, even colder.
[u'perfect', u'someone', u'(', u')', u'like', u'beer', u'ice', u'cold', u'case', u'even', u'colder', '<TERM>']
perfect someone ( ) like beer ice cold case even colder <TERM>
Service is quick and friendly.
[u'service', u'q

You need at least 3 mins to get to your phone book from the time you first turn on the phone.Battery life is short.
[u'need', u'least', u'3', u'min', u'get', u'phone', u'book', u'time', u'first', u'turn', u'phone', u'battery', u'life', u'short', '<TERM>']
need least 3 min get phone book time first turn phone battery life short <TERM>
Very Very Disappointed ordered the $35 Big Bay Plater.
[u'disappoint', u'order', u'$', u'35', u'big', u'bay', u'plater', '<TERM>']
disappoint order $ 35 big bay plater <TERM>
I mean this in a terrible way.  
[u'mean', u'terrible', 'way', '<TERM>']
mean terrible way <TERM>
It fits my ear well and is comfortable on.
[u'fit', u'ear', u'well', u'comfortable', '<TERM>']
fit ear well comfortable <TERM>
I couldn't think of a better way to end the film, but a smile on my face, great way to wrap up an amazing film and trilogy!  
[u'could', 'not', u'think', u'better', 'way', u'end', u'film', u'smile', u'face', u'great', 'way', u'wrap', u'amaze', u'film', u'trilogy',

[u'stanwyck', u'morgan', u'perfectly', u'cast', u'many', 'way', u'modern', u'equivalent', u'dickens', u"'", u'christmas', u'carol', u'sensibility', '<TERM>']
stanwyck morgan perfectly cast many way modern equivalent dickens ' christmas carol sensibility <TERM>
Awesome service and food.
[u'awesome', u'service', u'food', '<TERM>']
awesome service food <TERM>
Price is good too.
[u'price', u'good', '<TERM>']
price good <TERM>
Almost everyone involved must be return to school, acting is utterly predictable and bad, script is pile of garbage all round.  
[u'almost', u'everyone', u'involve', u'must', u'return', u'school', u'act', u'utterly', u'predictable', u'bad', u'script', u'pile', u'garbage', u'round', '<TERM>']
almost everyone involve must return school act utterly predictable bad script pile garbage round <TERM>
I have this phone and it is a thorn in my side, I really abhor it.
[u'phone', u'thorn', u'side', u'really', u'abhor', '<TERM>']
phone thorn side really abhor <TERM>
So I had to 

[u"'ve", u'buy', u'$', u'5', u'wire', u'headphone', u'sound', u'better', '<TERM>']
've buy $ 5 wire headphone sound better <TERM>
wont work right or atleast for me.
[u'wont', u'work', u'right', u'atleast', '<TERM>']
wont work right atleast <TERM>
I came over from Verizon because cingulair has nicer cell phones.... the first thing I noticed was the really bad service.
[u'came', u'verizon', u'cingulair', u'ha', u'nicer', u'cell', u'phone', u'first', u'thing', u'notice', u'wa', u'really', u'bad', u'service', '<TERM>']
came verizon cingulair ha nicer cell phone first thing notice wa really bad service <TERM>
Very very fun chef.
[u'fun', u'chef', '<TERM>']
fun chef <TERM>
The food was very good.
[u'food', u'wa', u'good', '<TERM>']
food wa good <TERM>
I like design and look of Jabra behing the ear headsets and 5020 is pretty comfortible to wear 24 hours a day without pain in the ear.
[u'like', u'design', u'look', u'jabra', u'behing', u'ear', u'headset', u'5020', u'pretty', u'comfortible', u'

This is by far the worst purchase I've made on Amazon.
[u'far', u'worst', u'purchase', u"'ve", u'made', u'amazon', '<TERM>']
far worst purchase 've made amazon <TERM>
Kieslowski never ceases to amaze me.  
[u'kieslowski', u'never', u'cease', u'amaze', '<TERM>']
kieslowski never cease amaze <TERM>
The burger is good beef, cooked just right.
[u'burger', u'good', u'beef', u'cook', u'right', '<TERM>']
burger good beef cook right <TERM>
My fiancé and I came in the middle of the day and we were greeted and seated right away.
[u'fianc', u'came', u'middle', u'day', u'greet', u'seat', u'right', u'away', '<TERM>']
fianc came middle day greet seat right away <TERM>
Love this headset!
[u'love', u'headset', '<TERM>']
love headset <TERM>
the charger worked for about a week and then completely stopped charging my phone.
[u'charger', u'work', u'week', u'completely', u'stop', u'charge', u'phone', '<TERM>']
charger work week completely stop charge phone <TERM>
The calls drop, the phone comes on and off 

[u'get', u'angry', u'want', u'damn', u'pho', '<TERM>']
get angry want damn pho <TERM>
So in a nutshell: 1) The restaraunt smells like a combination of a dirty fish market and a sewer.
[u'nutshell', u':', u'1', u')', u'restaraunt', u'smell', u'like', u'combination', u'dirty', u'fish', u'market', u'sewer', '<TERM>']
nutshell : 1 ) restaraunt smell like combination dirty fish market sewer <TERM>
Truly awful.
[u'truly', u'awful', '<TERM>']
truly awful <TERM>
I'll even say it again – this is torture.  
[u"'ll", u'even', u'say', u'torture', '<TERM>']
'll even say torture <TERM>
One of the worst shows of all time.  
[u'one', u'worst', u'show', u'time', '<TERM>']
one worst show time <TERM>
Worth every penny.
[u'worth', u'every', u'penny', '<TERM>']
worth every penny <TERM>
After a year the battery went completely dead on my headset.
[u'year', u'battery', u'go', u'completely', u'dead', u'headset', '<TERM>']
year battery go completely dead headset <TERM>
Schrader has made a resume full of lousy,

The service here is fair at best.
[u'service', u'fair', u'best', '<TERM>']
service fair best <TERM>
The service was extremely slow.
[u'service', u'wa', u'extremely', u'slow', '<TERM>']
service wa extremely slow <TERM>
I don't know what the big deal is about this place, but I won't be back "ya'all".
['not', u'know', u'big', u'deal', u'place', u'wo', 'not', u'back', u'``', u"ya'all", u"''", '<TERM>']
not know big deal place wo not back `` ya'all '' <TERM>
the worst phone from samsung...crap..... this crappy Samsung E715.. is the worst i have ever seeen.
[u'worst', u'phone', u'samsung', u'crap', u'crappy', u'samsung', u'e715', u'worst', u'ever', u'seeen', '<TERM>']
worst phone samsung crap crappy samsung e715 worst ever seeen <TERM>
You also cannot take pictures with it in the case because the lense is covered.
[u'also', u'not', u'take', u'picture', u'case', u'lense', u'cover', '<TERM>']
also not take picture case lense cover <TERM>
You get extra minutes so that you can carry out the call

I wasted my little money with this earpiece.
[u'waste', u'little', u'money', u'earpiece', '<TERM>']
waste little money earpiece <TERM>
All in all, I can assure you I'll be back.
[u'assure', u"'ll", u'back', '<TERM>']
assure 'll back <TERM>
Love this place, hits the spot when I want something healthy but not lacking in quantity or flavor.
[u'love', u'place', u'hit', u'spot', u'want', u'something', u'healthy', u'not', u'lack', u'quantity', u'flavor', '<TERM>']
love place hit spot want something healthy not lack quantity flavor <TERM>
However, this didn't make up for the fact that overall, this was a tremendously boring movie.  
[u'however', 'not', u'make', u'fact', u'overall', u'wa', u'tremendously', u'bore', u'movie', '<TERM>']
however not make fact overall wa tremendously bore movie <TERM>
There is nothing authentic about this place.
[u'nothing', u'authentic', u'place', '<TERM>']
nothing authentic place <TERM>
This is definitely a cult classic well worth viewing and sharing with others

In [17]:
curated.columns = ['Text', 'Sentiment']

In [18]:
curated

Text  Sentiment
0                    not make call certain place <TERM>          0
1     co-star not fare much better people like morga...          0
2     guy say better dialogue pot plant ha right <TERM>          0
3                                          10/10 <TERM>          1
4                  great restaurant mandalay bay <TERM>          1
5     `` 1 2 '' mega pixel camera part phone reasona...          1
6     saw `` mirrormask '' last night wa unsatisfact...          0
7              food wa excellent service wa good <TERM>          1
8     cashew cream sauce wa bland vegetable undercoo...          0
9                    horrible horrible protector <TERM>          0
10    however paul schrader ha indeed made film `` '...          1
11                      assure wo not disappoint <TERM>          1
12    want wait mediocre food downright terrible ser...          0
13    buy nokia cell phone week ago wa nightmare <TERM>          0
14    eew <TERM> location need complete overhaul <TERM>          0
15                           return 8 hour later <TERM>          0
16      not know pull pork could soooo delicious <TERM>          1
17                                       go rent <TERM>          1
18    anyways food wa definitely not fill price pay ...          0
19    level 5 spicy wa perfect spice not over-whelm ...          1
20    muddy low quality sound case around wire inser...          0
21                        good product well made <TERM>          1
22                      filmiing wa le expansive <TERM>          0
23                        o gosh best phone ever <TERM>          1
24    tracfonewebsite user friendly make easy toacti...          1
25    chosen torture disgust piece blatant american ...          0
26                                    place good <TERM>          1
27                     probably wo not come back <TERM>          0
28    film nothing pretext show ridiculous action sc...          0
29    wa look headset long time 've got could not ha...          1
...                                                 ...        ...
2370                      really good show watch <TERM>          1
2371                   buy different phone - not <TERM>          0
2372                  jabra eargels fit ear well <TERM>          1
2373  yet plantronincs continue use flaw charger des...          0
2374                 best charger see market yet <TERM>          1
2375               selection menu wa great price <TERM>          1
2376           love look feel samsung flipphones <TERM>          1
2377  main thing not enjoy crowd older crowd around ...          0
2378  fact hard remember part ray charles act not pl...          1
2379                     ha long bay wa bit flop <TERM>          0
2380           try cable computer ipod work fine <TERM>          1
2381  work well pair treo 700w skype headset pc ( us...          1
2382                  literally vomit watch film <TERM>          0
2383  want movie not gross give chill great choice <...          1
2384  cast also horrible cause see really really bad...          0
2385  movie wa bore sometimes find occupy peak paper...          0
2386  decor nice piano music soundtrack pleasant <TERM>          1
2387                                 worth drive <TERM>          1
2388                              disappointment <TERM>          0
2389                               know gave 10+ <TERM>          1
2390              wa shock no sign indicate cash <TERM>          0
2391     food wa good service wa good price good <TERM>          1
2392  worst software ever use <TERM> could give zero...          0
2393  pro : -good camera - nice picture also ha cool...          1
2394            wa deeply impress character play <TERM>          1
2395  choice headset year great sound good volume go...          1
2396  meat wa pretty dry slice brisket pull pork <TERM>          0
2397       'm really disappoint charger not work <TERM>          0
2398                           

In [19]:
def create_lexicon(data):
    """
    Function that creates a lexicon from the training data
    """
    vocab = {}
    excl = ['<TERM>', '<END>']
    for index, row in data.iterrows():
        text = row['Text']
        words = text.split()
        for word in words:
            if word not in excl:
                if word in vocab:
                    vocab[word][row['Sentiment']] += 1
                else:
                    vocab[word] = [0, 0]
                    vocab[word][row['Sentiment']] += 1
    return vocab

In [20]:
vocab = create_lexicon(curated)

In [21]:
def normalize(vocab):
    """
    Function that normalizes the vocabulary
    """
    for k in vocab.keys():
        total = np.sum(vocab[k])
        vocab[k] =  [x / (1.0*total) for x in vocab[k]]

In [22]:
normalize(vocab)

In [23]:
vocab

{u'secondly': [0.0, 1.0],
 u'uhura': [1.0, 0.0],
 u'ngage': [1.0, 0.0],
 u'magnetic': [1.0, 0.0],
 u'child': [0.0, 1.0],
 u'bear': [1.0, 0.0],
 u'yellow': [0.0, 1.0],
 u'sleek': [0.0, 1.0],
 u'four': [0.0, 1.0],
 u'catch': [0.5, 0.5],
 u'sleep': [1.0, 0.0],
 u'appetite': [1.0, 0.0],
 u'captain': [1.0, 0.0],
 u'hate': [1.0, 0.0],
 u'poorly': [1.0, 0.0],
 u'whose': [0.0, 1.0],
 u'over-hip': [1.0, 0.0],
 u'buddy': [0.0, 1.0],
 u'spew': [1.0, 0.0],
 u'sweetest': [0.0, 1.0],
 u'originality': [1.0, 0.0],
 u'brilliant': [0.0, 1.0],
 u'sorrentino': [0.0, 1.0],
 u'whatsoever': [1.0, 0.0],
 u'olde': [1.0, 0.0],
 u'lord': [0.0, 1.0],
 u'sorry': [1.0, 0.0],
 u'restock': [1.0, 0.0],
 u'worth': [0.391304347826087, 0.6086956521739131],
 u'slow-moving': [1.0, 0.0],
 u'louder': [0.0, 1.0],
 u'risk': [1.0, 0.0],
 u'appal': [1.0, 0.0],
 u'blanket': [0.0, 1.0],
 u'rise': [1.0, 0.0],
 u'relatively': [0.5, 0.5],
 u'distort': [1.0, 0.0],
 u'smack': [0.0, 1.0],
 u'every': [0.34375, 0.65625],
 u'jack': [0.0, 1

In [24]:
def predict(new_sample, vocab):
    """
    Function that predicts for a new sample, from the sum of the vocab vectors
    """
    c_sample = preprocessor(new_sample)
    curr_out = np.array([0, 0])
    for w in c_sample:
        if w in vocab:
            curr_out = curr_out + np.array(vocab[w])
    tot = np.sum(curr_out)
    curr_out = curr_out / tot
    return curr_out

In [25]:
predict('Very good product', vocab)

array([ 0.31552288,  0.68447712])

Not bad! Now let's test our estimator.
---

In [26]:
count = 0
for index, row in test.iterrows():
    text = row['Text']
    print text
    p = (np.argmax(predict(text, vocab)))
    if p!=row['Sentiment']:
        count += 1

The only place good for this film is in the garbage.  
The pizza selections are good.
Jawbone Era is awesome too!
The scripting of the subtle comedy is unmatched by any movie in recent years.  
After receiving and using the product for just 2 days it broke.
Would not go back.
Perfect for the PS3.
It also took her forever to bring us the check when we asked for it.
you can even take self portraits with the outside (exterior) display, very cool.
DELETE this film from your mind!  
He was very impressed when going from the original battery to the extended battery.
The ambience is wonderful and there is music playing.
The secondary plot line is incomprehensible and its relation to the primary plot line is mystifying.  
Sorry, I will not be getting food from here anytime soon :(
Hitchcock is a great director.  
Lame would be the best way to describe it.  
Her lines seem to have been WRITTEN by a fifteen year old, though they are trying oh so, so hard to sound like how a fifteen year old woul

The block was amazing.
No real improvement.
Very good food, great atmosphere.1
Works great, when my cat attacked the phone he scratched the protective strip instead of destroying the screen.
VERY comfortable.
The acting is like watching wooden puppets moving around and reading from a book, that's how bad it is.  
The cast of veteran actors are more than just a nostalgia trip.  
This is a great phone!.
Nice case, feels good in your hands.
Brilliant!  
Don't do it!!!!
Fans of the genre will be in heaven.  
I tried the Cape Cod ravoli, chicken,with cranberry...mmmm!
This was my first and only Vegas buffet and it did not disappoint.
Phone is sturdy as all nokia bar phones are.
I got the aluminum case for my new Palm VX and it worked really well--it has protected my handheld perfectly so far.
All I have to say is the food was amazing!!!
Hands down my favorite Italian restaurant!
The attractive set used throughout most of the film is an eye-pleasing gem.  
Nothing new there.  
The acting hel

VERY funny!  
None of the three sizes they sent with the headset would stay in my ears.
Makes it easier to keep up with my bluetooth when I'm not wearing it.
It will drive you barking mad!  
Only had this a month but it's worked flawlessly so far.
Works like a charm.. Works as advertised.
My boyfriend tried the Mediterranean Chicken Salad and fell in love.
Total brunch fail.
The salad had just the right amount of sauce to not over power the scallop, which was perfectly cooked.
Verizon's bills, however, are difficult to understand even though their pricing plans are simple.
I love the Pho and the spring rolls oh so yummy you have to try.
The worst piece of crap ever along with the Verizon customer service.
Not good for the money.
Three of the most visually appealing movies i've ever seen.  
Because both ears are occupied, background is not distracting at all.
Overall, I am psyched to have a phone which has all my appointments and contacts in and gets great reception.
But it is entertain

For those that haven't seen it, don't waste your time!  
I got food poisoning here at the buffet.
I could care less... The interior is just beautiful.
Boy was that sucker dry!!.
I work in the hospitality industry in Paradise Valley and have refrained from recommending Cibo any longer.
It's not what it says it is.
It felt too light and "tinny.".
The new characters weren't all that memorable, and I found myself forgetting who was who.  
Technically, the film is well made with impressive camera-work, solid acting and effective music from Riz Ortolani – particularly good is a recurring unaccompanied female vocal that sounds like it's coming from a distant hill.  
Will order from them again!
Unfortunately it's easy to accidentally activate them with the gentle-touch buttons if you accidentally touch the phone to your face while listening.
The lead man is charisma-free.  
Its well-designed and very sharp -- the blue is a very nice color.
the only VERY DISAPPOINTING thing was there was NO SPE

This show is made for Americans - it is too stupid and full with hatred and clichés to be admitted elsewhere.  
The only good thing was our waiter, he was very helpful and kept the bloddy mary's coming.
The burger... I got the "Gold Standard" a $17 burger and was kind of disappointed.
Excellent starter wireless headset.
It only recognizes the Phone as its storage device.
The folks at Otto always make us feel so welcome and special.
It seems completely secure, both holding on to my belt, and keeping the iPhone inside.
It was horrible!.
My sister has one also and she loves it.
It's like a bad two hour TV movie.  
This is simply the BEST bluetooth headset for sound quality!
It's dumb and pointless, and a complete waste of time.  
If you haven't choked in your own vomit by the end (by all the cheap drama and worthless dialogue) you've must have bored yourself to death with this waste of time.  
Definitely not worth the $3 I paid.
During several different 2 minute calls, I experienced 5 dro

In [27]:
print float(count)/test.shape[0]

0.206666666667


Not bad for a baseline, lets move on to word vectors and neural networks
---

In [28]:
word2vec = {}
with open('glove.6B.50d.txt', 'r') as f:
    for line in f:
        tabs = line.split(' ', 1)
        word2vec[tabs[0]] = np.array([float(x) for x in tabs[1].split(' ')])

In [29]:
word2vec

{'biennials': array([  7.53020000e-01,   8.01380000e-01,  -7.38770000e-01,
         -4.98160000e-01,  -1.04830000e+00,  -5.86180000e-01,
         -2.98530000e-01,  -1.42300000e+00,  -2.03960000e-01,
          1.54980000e+00,  -1.24250000e-02,   5.02400000e-01,
         -3.64180000e-01,  -4.30270000e-01,   7.28100000e-01,
         -5.63510000e-01,  -2.22240000e-01,   7.02190000e-01,
          1.08830000e-01,  -5.71190000e-01,  -1.41050000e-01,
         -5.45710000e-01,  -3.61990000e-02,  -2.06680000e-01,
         -7.77370000e-01,   1.19480000e+00,   6.10530000e-02,
         -1.78830000e-01,  -5.05890000e-01,  -5.20990000e-01,
         -1.10480000e+00,   7.11490000e-01,   4.37160000e-01,
         -8.73380000e-01,   5.57650000e-02,   8.47260000e-02,
         -2.53160000e-01,   1.12060000e+00,  -3.65580000e-01,
          6.80500000e-02,  -7.84590000e-04,  -1.02470000e+00,
          7.25030000e-01,   4.67220000e-01,   9.32570000e-01,
          9.60820000e-01,   1.16780000e-01,  -3.62180000e

In [30]:
def list2vec(tokens):
    out = np.zeros(50)
    for t in tokens:
        if t in word2vec:
            out += word2vec[t]
    return out

In [33]:
def normalized(a, axis=-1, order=2):
    l2 = np.atleast_1d(np.linalg.norm(a, order, axis))
    l2[l2==0] = 1
    return a / np.expand_dims(l2, axis)

In [74]:
# Batch preprocessor
def batch_preprocessor_NN(inp):
    data_ = []
    # Assuming that the columns are 'Text' and 'Sentiment'
    for index, row in inp.iterrows():
#         print row['Text']
        total = preprocessor(row['Text'])
        total = normalized(list2vec(total))[0]
#         print total
#         print '==========================================================================================='
        data_.append((total, row['Sentiment']))
    return pd.DataFrame(data_)

In [46]:
train_ = batch_preprocessor_NN(train)
train_.columns = ['X', 'y']
Xtrain = train_['X']
ytrain = train_['y']

I cannot make calls at certain places.
[  8.71151328e-02   4.84892793e-02  -3.19256198e-02  -1.10704005e-02
   1.10934804e-01   7.15567088e-03  -6.20677554e-02   2.93329662e-02
  -6.04020283e-02   5.29329044e-03  -3.54581265e-02   6.55568394e-02
  -1.85276005e-02  -2.02089551e-02   1.26975557e-01   1.02373563e-01
   7.15643871e-02  -9.21362069e-02   4.01075285e-04  -1.72385590e-01
  -5.10738663e-03  -8.96096133e-04   5.82123289e-02   5.56131984e-02
   4.70366461e-02  -3.75673851e-01  -9.37314748e-02  -2.53241103e-02
   1.35119012e-01  -9.83574356e-02   7.85335218e-01   9.60421743e-02
  -1.64437705e-01  -1.08563763e-01   2.62406215e-02  -3.72407887e-02
   6.83280392e-02   1.30022917e-02  -5.30779781e-02  -3.73844306e-02
  -1.99946418e-02   1.06493166e-02   2.98925746e-02   1.10021545e-01
   7.81911625e-03   6.10014733e-02  -4.89596395e-02   6.74465905e-02
   3.61482199e-02   5.69886879e-02]
His co-stars don't fare much better, with people like Morgan Freeman, Jonah Hill, and Ed Helms ju

[ 0.12294723 -0.05499185  0.01641752 -0.09992933  0.11382184  0.08540948
  0.05009371  0.05325588 -0.13838694  0.118623   -0.00684976  0.1305202
  0.04747561 -0.02005208 -0.01948238  0.15354374  0.0967121  -0.02800791
  0.12307833 -0.091054   -0.09178856  0.02979316  0.05934376  0.05323196
  0.15999912 -0.38642291 -0.18127205  0.01641802  0.16348817 -0.202029
  0.67060647  0.12506022 -0.08067758 -0.13367872 -0.04567574  0.02146694
  0.00067627 -0.05115925 -0.12205766 -0.14359377 -0.02000327  0.01077375
 -0.04552258  0.03401436  0.06856007 -0.01343277 -0.01848057  0.00289196
  0.01212838  0.068664  ]
I consider this theft.
[ 0.05825493 -0.03054816  0.01426466  0.06326975  0.08428825  0.14130198
  0.08574384 -0.02932017  0.05938732  0.11273161 -0.05515463  0.02555912
 -0.07620392 -0.10993102  0.29210347 -0.02962374  0.0050641  -0.04625301
  0.03588746 -0.12758261  0.20689273 -0.13175372 -0.11023215  0.02263973
 -0.1025997  -0.45127797 -0.05743379 -0.10136401  0.2056097  -0.07970914
  0.4

Good Service-check!
[ -6.75267217e-02   9.89200248e-02  -1.15884249e-01  -5.71755087e-02
   1.80006741e-01  -5.98472791e-02  -1.13533167e-01   2.31275132e-02
  -6.06138951e-03   1.05684841e-01  -2.01540300e-02   1.20303676e-01
  -8.98306920e-02  -1.44015639e-02   7.25761402e-02   1.54782419e-02
   1.56006348e-01   4.21638778e-02  -1.58947573e-03  -1.45391373e-01
  -1.06743682e-01   1.17191671e-01   3.85053739e-02  -9.22178278e-03
   1.66634605e-01  -3.14027909e-01  -1.46905630e-01   2.92889043e-02
   1.79932736e-01  -7.49917395e-02   7.07829420e-01   1.57222687e-01
  -2.67632463e-02   3.11105660e-03   4.00670693e-02  -6.84736063e-03
  -2.95773341e-02   1.64296806e-01   4.99230181e-02  -1.34755526e-01
  -6.97734378e-03   3.46912698e-04  -3.35944776e-02   5.12949570e-02
   2.09225435e-02   2.68182757e-02  -1.08772178e-02   5.16270308e-02
   5.94032491e-02   1.76038934e-01]
Very disappointing.  
[-0.02511314 -0.18146151  0.20439633 -0.13615909  0.09822045 -0.22835796
 -0.01351274  0.19815

Now this dish was quite flavourful.
[ 0.03344493 -0.01498799 -0.09836074  0.13736256  0.09235857 -0.09001988
 -0.09633334 -0.00952694 -0.17656342  0.07457593 -0.01433268  0.0738649
  0.25689712  0.06495265 -0.1069495   0.12228633  0.02747867  0.13741796
  0.06581788  0.05548984 -0.10348773 -0.00943224  0.23264321  0.12035332
  0.14576868 -0.19747048 -0.24402067  0.16520854  0.1384922  -0.10624155
  0.58008595 -0.01039392 -0.07590133  0.14088383  0.05296585  0.02974092
 -0.03646982  0.20249384 -0.04151452 -0.14561273  0.08783488  0.03048488
 -0.10891472 -0.03896488  0.05451205  0.15019139 -0.07143315  0.05917753
  0.14944478  0.10694744]
I just got bored watching Jessice Lange take her clothes off!  
[ 0.11279847 -0.05393882  0.12202167 -0.13634077  0.22044346  0.08673732
 -0.12784539  0.04794007  0.030363   -0.07555799 -0.02537649  0.05754716
 -0.05035531 -0.00492073  0.15989946  0.04376705  0.07624364 -0.01654866
 -0.0608952  -0.21884084 -0.01316403  0.14811685  0.12237238 -0.01888855

Battery is terrible.
[ 0.07386586 -0.01623084  0.09241066  0.00528156  0.01625404  0.07663798
  0.2836056   0.18692275  0.05565974  0.01672979 -0.06855443  0.0546277
 -0.13492197 -0.1224973   0.04113511 -0.06213775 -0.11893714  0.17474316
 -0.13233506 -0.07263127  0.01634471 -0.02686516 -0.02108838 -0.09943763
 -0.00756033 -0.36478911 -0.192719    0.01328191  0.41459126  0.01265668
  0.46915362  0.01374459  0.04885731 -0.12598278  0.09217332  0.21841345
  0.18131317 -0.07565621  0.08217422 -0.06340648 -0.02278099 -0.04382203
  0.00481354  0.0122648   0.11173417  0.06158533  0.01130716 -0.09913762
  0.07575368  0.11994652]
To summarize... the food was incredible, nay, transcendant... but nothing brings me joy quite like the memory of the pneumatic condiment dispenser.
[ 0.01121503  0.01529232 -0.05545807  0.04223594  0.02255861 -0.04989992
  0.05561614 -0.0221701   0.03634103  0.21811089 -0.0287392   0.14446368
  0.03821468 -0.08620144  0.07239772  0.11192143  0.00694839  0.08903928
  0

[ 0.07371945  0.02019257  0.10151899 -0.02500967  0.09296812  0.05587815
 -0.12801258 -0.0483504   0.00662493 -0.02188301 -0.03509693  0.08058152
 -0.13845638 -0.05790001  0.13488621  0.10467474  0.04699793  0.01203976
  0.05178744 -0.13599609 -0.0218397   0.00203495 -0.00874975  0.01028078
  0.05335058 -0.4179992  -0.10485804  0.07555037  0.19766294 -0.13632232
  0.74560094  0.10135917 -0.07016408  0.00764367 -0.05128998 -0.04946326
 -0.00642799  0.01770105  0.05552285 -0.07633393 -0.00637791 -0.00116834
 -0.01731149  0.10182475 -0.08422117 -0.03950806 -0.00453946  0.04827399
 -0.01773455  0.09563449]
The movie is not completely perfect but 'Titta Di Girolamo' will stay with you for a long time after the vision of the movie.  
[ 0.15639929  0.03657123 -0.08187275 -0.04843819  0.06587604 -0.03219702
 -0.09534361 -0.08911086 -0.07342687  0.12120718 -0.02856401  0.10813708
 -0.1154632   0.05248712  0.17810979 -0.00469397  0.03298523 -0.00349301
 -0.05387086 -0.02702893 -0.08439519  0.074

[-0.09168054 -0.06556672 -0.02491514  0.06316677 -0.02685    -0.20798843
 -0.1514741  -0.04691567  0.03749076  0.01231049 -0.03512957  0.07429167
  0.06581732 -0.13341425  0.04067327 -0.03575862 -0.09993497  0.06982637
  0.10602268  0.16647155 -0.04781154  0.02172325  0.1210137   0.06817671
  0.00313132 -0.26466497 -0.24160605 -0.02308266  0.15533673 -0.02771312
  0.66248584  0.11413727 -0.03411209  0.02287958  0.06082937  0.14741729
  0.04890051 -0.08262928  0.06689578 -0.06815316 -0.03195487 -0.02974333
 -0.01361017 -0.10412819  0.08189689  0.13710547 -0.14766254  0.13311144
  0.25330256  0.10469176]
Absolutel junk.
[ 0.07002689  0.03329891  0.25088708 -0.20138125  0.02936726 -0.15222677
 -0.08993641 -0.14759018  0.12207006  0.2205369  -0.14516626  0.31833439
  0.07465458  0.16497349  0.07728309 -0.02942286 -0.10816031 -0.00177613
  0.0517652  -0.03940764  0.16690151 -0.14304255  0.10125102  0.06850583
 -0.08956504 -0.28184212 -0.0989027   0.14179056  0.1786364  -0.07236854
  0.26723

[ 0.06923371  0.04119589  0.03691113 -0.02597206  0.0021467   0.03333406
 -0.12834982 -0.07716733  0.02373222  0.05184156 -0.04925339  0.04231921
 -0.06906427  0.02170689  0.04904491  0.05584606  0.04211969  0.07511722
  0.01299296 -0.25695735  0.04139374  0.04224245 -0.04160363  0.06621899
  0.07928126 -0.33479906 -0.10553322  0.04956139  0.21194052 -0.0678794
  0.7806274   0.10364789 -0.09043968  0.023296    0.09904904  0.03111423
  0.02768434  0.11333276 -0.00883034 -0.048109    0.0018256   0.05381572
  0.00761959  0.07644433  0.01713465 -0.018958    0.04624221 -0.03874175
  0.05143232  0.11665554]
All in all a beautiful directed film from Nicola's roeg wih a sublime cast.  
[ 0.18111242  0.08299437 -0.20183956 -0.13286933 -0.03698362  0.10295373
 -0.00379255 -0.13968817 -0.06730211  0.29113795 -0.07186778  0.15601077
 -0.05515027  0.22795834  0.02241909 -0.11958606  0.12305813 -0.09685115
 -0.09636867  0.03459791  0.04163952  0.20268024 -0.02452961 -0.01232464
  0.14133255 -0.20655

[ 0.0163425   0.0744183  -0.11709446 -0.12054988  0.09989241 -0.02130855
 -0.14518329 -0.02193747 -0.00332633  0.0587616  -0.00616234  0.11412706
 -0.09823837  0.00357301  0.08037487 -0.03677668  0.12661006  0.05111006
  0.04332203 -0.14221588  0.02579075  0.11126637  0.00783918 -0.02832071
  0.15736997 -0.30374478 -0.1550522  -0.05611031  0.10115242 -0.06485194
  0.78016696  0.07276571 -0.06431212 -0.10188288 -0.00089209  0.0486519
 -0.01346078  0.17313017  0.04790746 -0.04374335 -0.02029939 -0.03134757
 -0.00173962  0.00905764  0.00742857 -0.00361186 -0.00313743  0.03304266
 -0.03273266  0.13313649]
Good value, works fine - power via USB, car, or wall outlet.
[ 0.04183854  0.09775702  0.11978166  0.03026336  0.04625112  0.01108422
 -0.08813037 -0.11549811  0.02724955  0.10623173  0.00597933  0.03621168
 -0.07895288  0.0018385   0.01728322  0.05980826 -0.01223157  0.06703108
 -0.01852273 -0.15819823  0.12953571 -0.05400175 -0.08556262  0.03383762
  0.02021562 -0.3711608  -0.05751509  

[  6.84401529e-02   3.41681606e-02  -2.56740220e-02   2.46585294e-03
  -3.76695677e-02   8.08605557e-02  -8.72089858e-02  -6.98235212e-02
  -2.81783639e-02  -6.13179703e-04  -1.99023324e-02   1.60244803e-01
  -3.91833563e-02  -2.54225514e-02  -9.77039170e-03   6.20896005e-02
  -1.72801913e-03   5.77978231e-02   4.81547684e-02  -9.93304669e-02
   8.98540715e-02   4.60560304e-02   5.84833206e-03  -6.55092796e-02
   9.66295611e-02  -3.08494301e-01  -1.26030977e-01   4.52919283e-02
   1.40020244e-01  -1.11007670e-01   8.35898120e-01   1.03028966e-01
  -2.12476598e-02  -2.88794706e-02   7.20290941e-03   1.60149460e-01
   3.53897949e-02   1.50051797e-02   1.64861370e-02  -7.77091251e-02
   6.25516977e-03  -4.60355284e-03  -3.60662107e-02  -4.13701090e-02
   1.86909090e-03   7.35751575e-02   1.95168777e-02   3.04720641e-02
   1.63597610e-02   3.80056893e-02]
The phone gets EXTREMELY HOT!
[  4.73405761e-02   9.76756918e-03   1.19738338e-01   3.75181773e-02
   2.75633083e-04  -2.72978927e-02  -

[ 0.08414899  0.02466558 -0.01481614 -0.04623827  0.10627854  0.04052543
 -0.06527498 -0.01200307 -0.09513815  0.03586588 -0.03070477  0.03480435
 -0.10914386  0.0368262   0.0451937   0.1007505   0.07078877 -0.01904165
  0.02422328 -0.13527032 -0.03038401  0.08643329  0.03470043  0.08988578
  0.11122405 -0.36593813 -0.09557242  0.05788496  0.1144475  -0.10736516
  0.820612    0.03701236 -0.07191595 -0.08607236  0.02876653  0.00494003
  0.02042291  0.02560495 -0.09904485 -0.04121558 -0.00193476  0.02809871
 -0.02943     0.06854652 -0.00186445  0.01917972  0.03433547  0.00758535
 -0.01857645  0.0152027 ]
Nice design and quality.
[ -1.68570340e-02   1.23952208e-01  -1.40582285e-01  -4.37085139e-02
   4.13684423e-02  -7.11651619e-03  -9.94967621e-02  -1.76877606e-01
   1.90449974e-02   9.92907905e-02   4.53277623e-02   5.61471054e-02
  -3.85555686e-02  -2.34038247e-02  -2.45197249e-02   1.32964441e-02
   3.72584133e-02   8.71679963e-02   2.33564124e-02  -2.82953803e-01
   1.55419242e-02   

[  7.66403653e-02   1.14914837e-01  -5.68887195e-02  -1.65268672e-02
   1.34229987e-01  -5.88605678e-02  -1.49458034e-01  -2.28618959e-02
  -2.15264706e-02   2.82496845e-02  -1.18803860e-01   4.33851439e-02
   3.44714933e-03   1.00182441e-01   7.95990340e-02   5.11336114e-03
   5.19366914e-02   3.96189136e-02  -3.00679975e-02  -2.08829490e-01
   2.84932078e-02   5.60220025e-02   3.19732511e-02   8.93638956e-02
   6.55496149e-02  -2.61018040e-01  -1.15072585e-01   3.70905560e-02
   1.00481804e-01   3.21769789e-04   7.72588086e-01   1.48921422e-01
  -2.31780907e-02   1.33248007e-01   1.01075151e-01  -2.47000887e-03
   2.20892720e-02   1.71942482e-01   5.73252150e-03  -1.53434893e-02
  -1.09096988e-02   1.01457869e-01  -3.70058561e-02   9.96921681e-02
   1.32230354e-03   7.55282428e-02  -7.50291860e-02  -5.58499139e-02
   4.62793690e-02   1.55875504e-01]
The pan cakes everyone are raving about taste like a sugary disaster tailored to the palate of a six year old.
[  1.78087710e-02   6.048

[  5.29095765e-02   1.10675835e-01   2.75295342e-02  -2.46938306e-02
   1.13714747e-01  -5.42379002e-03  -1.67361205e-01  -3.92100544e-02
  -8.94616401e-02  -1.79962249e-02  -6.55680471e-02   6.93382713e-03
   9.42328528e-03  -4.84605300e-04   1.91001575e-01   2.07450054e-02
  -1.60289218e-02   2.51691154e-02  -1.10645604e-01  -1.12460845e-01
   6.10782134e-02  -4.02311217e-02   1.24073462e-01   1.02693607e-01
  -1.46760638e-02  -2.98817254e-01  -9.67040086e-02   2.92461282e-03
   1.30343555e-01  -8.68581379e-02   7.91370865e-01   1.19853087e-01
  -7.93936583e-02   5.91869314e-02   3.22824281e-02  -5.00357429e-02
   2.66004337e-02   8.96161802e-02   7.82420064e-02  -7.71033615e-02
  -4.42576992e-02  -2.93016942e-02   3.02820580e-02   7.41808116e-02
  -2.44216610e-02   9.02663914e-02  -5.74349920e-02  -9.33415766e-02
   3.71343872e-02   6.90490478e-02]
A very, very, very slow-moving, aimless movie about a distressed, drifting young man.  
[ 0.07380405  0.02318837 -0.04288524 -0.21357471

[-0.02898914  0.09770966 -0.16830222 -0.05732237  0.0793614   0.1169007
 -0.09651842 -0.14269167 -0.07873548  0.07864775 -0.0353144   0.12853437
  0.00107767 -0.00682466  0.12214216 -0.0771295   0.06428037  0.06259504
 -0.08706038 -0.23253682  0.08856825  0.04702244 -0.0392062  -0.07764742
  0.08990549 -0.25311607 -0.28875357  0.00105642 -0.03619985 -0.18294356
  0.63637906  0.02682477  0.03071858 -0.10678014  0.05655744  0.14503446
  0.07385384  0.1550742  -0.07172469 -0.13079369  0.06892043  0.15089833
 -0.03923204 -0.13921829 -0.03741361  0.03264096  0.02027479 -0.13773972
 -0.0616924   0.10738352]
I had Verizon 2 years ago and really liked their service.
[  5.91554120e-02   6.46282534e-02   1.02271756e-01   7.69803154e-02
   1.91196344e-02   1.86922021e-02  -2.06800382e-01  -6.89905974e-02
  -1.56436721e-02  -3.09137700e-02  -2.20065903e-02   9.28987850e-05
  -9.72120223e-02  -6.76523375e-02   1.68527202e-01   3.12033328e-02
  -1.05643923e-01   3.16786858e-02  -1.24120686e-01  -7.2

I used bitpim (a free program you can find on the internet)to transfer data to the phone.The price of the cable was excellent.
[  4.05127057e-02   6.29424852e-02   1.57346003e-01   8.52196248e-02
   1.02639091e-03  -2.64741014e-02  -1.40346558e-01  -1.59248680e-01
   7.44093007e-02   5.65493536e-02   5.78701369e-02   8.44527083e-02
   2.18633159e-02   9.60331425e-04   5.33755987e-02   6.16169269e-02
  -6.12758327e-02   1.49043710e-02  -6.70379258e-03  -4.34724840e-02
   1.01378640e-01  -3.40054715e-02   3.23874215e-02   9.26459741e-02
  -1.52905196e-03  -3.12317240e-01   5.19123178e-04  -8.41537576e-02
   6.25428069e-02  -6.22176871e-02   8.30910455e-01   9.43984387e-03
  -7.86804534e-02  -4.42414115e-02   6.50289849e-04   7.99515697e-03
   2.68867094e-02  -7.51832433e-03   1.17008855e-02  -6.13162283e-02
   1.57903876e-01   2.38024558e-02   2.61193865e-02   5.43503966e-02
  -4.34811922e-02   7.51005940e-04   4.42013643e-02   7.32898180e-02
   5.02063536e-02   8.14079793e-02]
much bett

It is an excellent drama, an excellent thriller, and an excellent film.  
[-0.06657928  0.1495258  -0.19585024 -0.0151208   0.06006809  0.00221009
 -0.02790234 -0.10268584  0.04671307  0.20304872 -0.04235244  0.09817995
 -0.04438548  0.15663003  0.05250313 -0.17723679  0.11939754  0.12818217
 -0.19643161 -0.0137897   0.07530292  0.16450843  0.01149143  0.0254945
  0.12778014 -0.20777698 -0.22667941  0.01749656  0.00766953 -0.00809248
  0.61030721 -0.0091175   0.11087734 -0.1428332   0.05420186  0.16284916
  0.00092753  0.12811164 -0.11014276 -0.19990079  0.06977283  0.09296286
 -0.00445183 -0.12176978 -0.09155266  0.01479922  0.11682522  0.02128227
  0.04482427  0.18328949]
Awsome Device Works Great!!.
[ 0.09661463  0.06033346 -0.01414799 -0.05459135  0.07282029  0.06280821
 -0.06106572 -0.0520896   0.0495544   0.04555114  0.04888991  0.09147191
 -0.03974685  0.0109394  -0.00561019  0.01491519  0.02484833  0.11830575
 -0.08761921 -0.12349827  0.08958011  0.0360535  -0.09401655 -0.05412

I found a six inch long piece of wire in my salsa.
[ -8.76934668e-04   8.76973013e-02   5.44578226e-02   5.76739820e-03
   8.25799651e-02   1.12365467e-01  -4.29169550e-02  -9.97181164e-03
  -5.51923228e-02  -1.40137795e-02  -1.05367165e-01  -5.02123924e-02
  -6.77298723e-03   1.78842747e-01   4.83702065e-02  -4.65755959e-02
  -3.63555425e-03   1.53711409e-02  -5.19406547e-02  -2.01023143e-01
   4.06371246e-02  -1.08160335e-02   1.24509824e-01   5.54919305e-03
  -1.80631366e-02  -2.35128333e-01  -8.56016850e-02   1.20136135e-01
   1.60840722e-01  -1.61192375e-01   7.94365807e-01  -5.89047142e-02
  -2.85232038e-02   1.28724024e-01  -2.73289669e-03   9.61843353e-02
   2.11694545e-04   8.95040543e-02   3.44789702e-02  -1.00224945e-01
   6.42657907e-02   1.09132133e-02   4.52700797e-02   1.27571926e-02
   4.69295652e-02  -1.05116327e-02   9.87602185e-02  -1.07198403e-01
  -1.14302353e-01  -3.42728676e-02]
Perhaps I caught them on an off night judging by the other reviews, but I'm not inspi

I even fully charged it before I went to bed and turned off blue tooth and wi-fi and noticed that it only had 20 % left in the morning.
[  4.57075842e-02   4.23124952e-02   8.88599144e-02  -4.36412265e-02
   1.30771210e-01   4.32466881e-02  -1.07082082e-01  -4.29086758e-02
  -2.32158254e-02  -6.07307719e-02   1.88681175e-02  -3.69485830e-02
  -1.20565290e-02  -1.69325588e-02   7.50656907e-02   1.11105928e-01
  -6.74687453e-02  -1.41158147e-02  -1.09706736e-01  -1.18682890e-01
   3.16998345e-02   8.53986473e-02   1.28915433e-01   6.28071455e-03
  -1.33911211e-02  -3.09560054e-01   2.26432981e-04   7.15112985e-02
   1.28056264e-01  -6.49430768e-02   8.35173071e-01   9.42269167e-02
  -6.36751191e-02  -1.59873363e-02   3.81182773e-02   1.05525555e-02
   8.16181681e-02  -6.64791661e-03   5.30832655e-02  -7.98679298e-02
  -2.29973943e-02  -2.52842679e-02   4.99089000e-03   9.04518659e-02
  -1.55251708e-02  -3.16282107e-02  -1.25159262e-02  -9.28554509e-02
   6.06735715e-02  -2.94873052e-02]


[ 0.10922267  0.09152224  0.07119361 -0.0554966   0.11350253 -0.07356877
 -0.11780614 -0.030223    0.04057505  0.0992685  -0.10291662  0.09706852
 -0.00889315  0.06860651  0.1175878   0.11853056  0.05582267 -0.14516808
  0.13250564 -0.14544661  0.07019513 -0.01322142  0.04724925  0.02757909
 -0.05357905 -0.4276842  -0.22190577  0.04299799  0.11036211 -0.00347806
  0.59131449  0.05934795 -0.0837235   0.00732189  0.14902031 -0.10243309
 -0.06456509  0.10358075  0.01823392 -0.10664928 -0.00828383  0.1018277
  0.11893068  0.11514248  0.10976576 -0.09314101 -0.02755552  0.19502809
  0.11016086  0.15868356]
Very good, though!
[  3.76379557e-03   3.81089765e-02  -7.49704581e-02  -7.55731681e-02
   1.37613502e-01   7.99314773e-03  -1.49132203e-01   2.59236338e-02
  -5.56780551e-02   8.41564580e-02   1.06269303e-02   8.50984998e-02
  -7.89664856e-02  -4.69113840e-02   1.19737659e-01   5.13094731e-02
   9.16494056e-02  -5.08287258e-03  -3.39983522e-02  -1.47927876e-01
  -1.03926220e-01   7.49053

first time there and might just be the last.
[  5.77284187e-02   2.95754865e-02   1.99991580e-02   1.67511928e-02
   5.62897535e-02   1.55867779e-02  -1.97090064e-01   5.85575890e-02
  -7.00983605e-02  -2.79278536e-03   4.82020402e-03  -2.49657901e-02
  -1.47280240e-01   1.04276865e-02   2.27316170e-01   3.57807576e-02
  -2.95990928e-02  -8.29324787e-02  -2.08462661e-01  -3.85785214e-02
   6.60913207e-03  -2.64376321e-04   4.93457190e-02  -3.17970755e-02
   2.69930984e-02  -4.08878127e-01  -5.45095404e-02  -4.84142588e-02
   3.85592037e-02  -8.75483016e-03   7.66189072e-01   5.74132645e-02
  -8.92162457e-02  -3.61951605e-02   4.83130341e-02  -5.57946633e-02
   5.16549385e-02   8.21995105e-02  -2.68212262e-03  -9.95765963e-02
  -9.48117513e-02   8.58294692e-03  -3.38387893e-02   1.92520051e-02
  -2.23923985e-02   4.20846260e-02   4.02046289e-03   1.03873953e-02
  -2.71607759e-02   3.03508432e-03]
About half way through this film I started to dislike it.  
[ 0.05166866  0.00178471 -0.048

Ray Charles is legendary.  
[ -3.86725985e-02   2.58322158e-01  -2.83199762e-04   2.77509381e-02
  -1.49682507e-03   1.00263270e-03  -2.59238599e-01  -2.92689592e-02
  -7.43504914e-02   5.24429670e-02   2.59365247e-03   2.97249812e-01
  -1.52130163e-01  -2.26735710e-02   2.66915072e-02  -2.80112708e-02
   2.05926684e-02  -3.53260920e-02  -2.34397055e-01  -1.03486646e-01
  -7.22942149e-02   9.52105285e-02  -9.39660327e-03  -1.25139994e-01
  -3.27816916e-02  -3.20064103e-01  -1.94928507e-01  -1.29577171e-01
  -4.83579424e-02   8.43609873e-02   3.00323673e-01  -1.34246361e-01
  -2.44519247e-02  -1.19855064e-01   2.03822035e-01   1.29383593e-02
   2.66946558e-02   4.05918485e-02   3.91061932e-02   5.06751673e-02
   6.39512555e-02   3.45311098e-01  -1.80020062e-01  -6.89762043e-02
   8.64550825e-03   1.57782724e-02  -7.85105376e-02  -3.12470480e-01
  -1.32977240e-01   1.49565534e-01]
Those burgers were amazing.
[ 0.10625688 -0.24295057 -0.04996054 -0.03285796  0.00640237 -0.06136237
  0.067

[-0.09620146 -0.07382846  0.00443735 -0.07735571  0.08815867  0.04046427
 -0.10862619 -0.07873878  0.05139172  0.14716336 -0.0111755   0.04697175
 -0.15097975  0.11002357  0.16018343 -0.05599027  0.091335   -0.0907542
 -0.30506643 -0.12556686  0.08683871  0.02972299  0.05934665 -0.03797288
  0.08625712 -0.34428343 -0.08073024  0.12124332  0.06202698 -0.14623911
  0.64396547  0.01898408  0.07577914  0.023434   -0.00517219  0.03971685
  0.03802019  0.09067333 -0.01433532 -0.13564214  0.00421061  0.20133814
 -0.04257212 -0.10246531 -0.10019555 -0.04492126  0.13423643 -0.11140838
  0.06108072 -0.00576494]
It'll be a regular stop on my trips to Phoenix!
[ 0.04085121  0.14812234 -0.00156451  0.01340592 -0.02756961 -0.11625265
 -0.2389086   0.0995954   0.03308882 -0.01068201 -0.11511962 -0.01242285
 -0.09510369  0.05454998  0.15348065 -0.00953764 -0.00166009 -0.01279678
 -0.1451157  -0.07719098  0.03517135  0.03832779  0.02773295  0.08092949
  0.03846705 -0.36433714  0.03412061  0.02242526  0

[  9.73948257e-02   6.33145915e-02   4.50150059e-02  -3.90215082e-02
   9.53304717e-02   8.78243790e-02   3.03758637e-02   2.96144167e-02
   1.22961147e-04   8.82394951e-02  -5.83871497e-02  -1.74801536e-02
  -3.68378265e-02   2.04371349e-02   1.38787978e-01  -1.65815374e-02
  -8.31399778e-03  -9.78860439e-03  -1.23500678e-01  -1.56320128e-01
   1.09153014e-02   4.53813585e-02   1.07292434e-01  -1.79361648e-02
   1.11020649e-01  -3.48974610e-01  -9.19552421e-02   9.49721879e-02
   1.89967354e-01   1.64440174e-03   8.01462360e-01   7.72640788e-02
  -2.54535087e-02  -1.02879164e-01  -8.05281497e-02   2.81476283e-02
   5.99872734e-02  -1.12522605e-02   1.64861154e-02  -9.08526763e-02
   2.81855440e-02   3.06386502e-02  -1.38592523e-02   1.92358093e-02
   5.37981705e-03   6.53571698e-02  -1.23572319e-02   1.99333876e-03
   5.96184651e-02   2.49919095e-02]
The owners are super friendly and the staff is courteous.
[ 0.00592835  0.06238594 -0.09515061  0.14481679  0.12612905 -0.14611429
 -0.2

This is one of Peter Watkins most accessible films.  
[ 0.04881853  0.15662371 -0.0264653  -0.02366705  0.08245508  0.07123472
 -0.20289707 -0.08465648  0.05136528  0.003424   -0.07344096  0.12365981
 -0.1441232   0.0360375   0.16887871 -0.0356822   0.15421634  0.03031297
 -0.13566866 -0.06688909  0.12493951  0.12458206  0.02034252  0.15002375
  0.1270816  -0.33459278 -0.12636504  0.00503313 -0.10046238 -0.08848941
  0.63485118 -0.1344497   0.10836401 -0.16209656  0.00407241 -0.00432157
  0.1223057   0.07853307 -0.00376532 -0.0595011   0.0819351   0.14738025
 -0.03098883 -0.04541153 -0.07269081  0.02303117 -0.06561241 -0.1811828
  0.01776965  0.11917002]
Great Phone.
[ 0.09499657  0.14847691  0.085401    0.10302489  0.08019216 -0.03371116
 -0.15698279  0.05850786 -0.01136223  0.01962046 -0.02018477  0.11636641
 -0.00962645 -0.04520816  0.04465767  0.01527634 -0.13570051  0.0516092
 -0.03884398 -0.09879861  0.08284957  0.05768576 -0.07022994  0.05015322
  0.20774651 -0.42002455 -0.16207

[ 0.01419287  0.07388532  0.0725617  -0.02545054  0.1535333   0.03660465
 -0.07377447 -0.06650722 -0.06797769  0.02918617  0.01215002  0.08633999
 -0.09453952  0.11792514  0.11110271  0.11373436  0.02749019  0.00720396
 -0.08078748 -0.08479983 -0.00757195  0.12261412  0.03726986  0.02147564
  0.05583124 -0.43486375 -0.1222608   0.00643325 -0.04019777 -0.05979226
  0.72911825  0.04591635  0.06172966 -0.13781893  0.01575877  0.03818617
  0.03172291  0.01047827 -0.09603856 -0.15777291  0.00949193  0.17190009
 -0.04679294  0.02437531  0.03158275 -0.06849638 -0.01485953 -0.0383911
 -0.07615812  0.09750434]
The refried beans that came with my meal were dried out and crusty and the food was bland.
[-0.04628303 -0.10592108 -0.20184974 -0.04168743  0.09710683 -0.05293257
 -0.19149629 -0.07848693 -0.01997031  0.03665798 -0.05553621  0.08277537
  0.21551674 -0.0273112   0.11528292  0.04849828 -0.00891536  0.03464222
  0.0730547  -0.12415456 -0.07400941 -0.02872245  0.23763071  0.05006235
 -0.0075

[-0.00575743  0.07026295 -0.07039731 -0.04471123  0.11117616 -0.07985138
 -0.18119659 -0.03763217  0.15549128 -0.01314247  0.0060988   0.09437676
  0.02071932 -0.03319167  0.08977846  0.01092814  0.00328418  0.05127287
  0.02519261 -0.13177587  0.04499219  0.0660478   0.02230656  0.03540233
  0.07016354 -0.31846062 -0.09943162 -0.015377    0.17901346  0.00903733
  0.78928862  0.146179   -0.01819837 -0.00982275  0.00106672  0.05191574
 -0.01189665  0.09908943  0.06598843 -0.04570169  0.05069718  0.019819
  0.0234645   0.03877584  0.0490373   0.05712261 -0.07255882  0.06501948
  0.05239874  0.16710151]
Think it over when you plan to own this one!This sure is the last MOTO phone for me!
[  1.15519383e-01   1.92792834e-02   1.42033960e-01   8.37012143e-02
   7.56225730e-02  -4.47461946e-02  -1.62357797e-01  -1.72398053e-02
  -5.02696191e-02   3.38291759e-02  -5.07193575e-02   1.62929548e-04
  -1.43053271e-01   4.31886208e-02   1.82565495e-01   4.83253861e-02
  -2.73608068e-02   5.93531845e

[ 0.0199473   0.19326633 -0.16485997 -0.00456812  0.17996398  0.04721674
 -0.0857819   0.05404631 -0.06008991  0.05470193 -0.06684923  0.05734156
 -0.12336188 -0.02578689  0.16197234 -0.01594992  0.09656805 -0.00239504
 -0.10190179 -0.04671383  0.00265322  0.12034062  0.07060134  0.12334715
  0.1296739  -0.29988335 -0.17311973  0.00201908  0.18076685 -0.14228775
  0.68803762  0.10588208 -0.11509985 -0.00114826 -0.02447013 -0.00250261
  0.00555784  0.04617719 -0.02509601 -0.11970194 -0.00314242 -0.07170725
 -0.02765037 -0.03259173  0.00491803  0.10475081  0.01545891 -0.2297977
  0.02417457 -0.00242674]
Barney is definitely super DE duper.  
[-0.01131118  0.00920018 -0.09650532  0.17793732  0.0665467  -0.2465855
 -0.07043069 -0.02168666 -0.05484681  0.11184748  0.02015306  0.20997382
 -0.23482149  0.03234938  0.11941484 -0.01416517  0.05208243 -0.0698555
 -0.30325498 -0.15447335 -0.18742034 -0.01887204  0.05706989  0.0350174
 -0.13240464 -0.23995648 -0.19104978  0.05053519  0.08206889  0

[ 0.01765178  0.03786854  0.01115172 -0.02966459  0.10067793  0.05150945
 -0.12801181  0.0191173  -0.14701773  0.08614308 -0.01046273  0.15531197
 -0.09209278  0.05236278  0.18062709  0.15757524  0.13820589 -0.03966281
  0.04965625 -0.02610102 -0.04244386  0.10171328  0.12038244  0.04582064
  0.11073697 -0.44781831 -0.1351629  -0.00299322  0.05023062 -0.15610216
  0.65964245  0.04289237 -0.16032552 -0.10797337  0.06695801 -0.01059473
  0.04056212  0.04605273  0.03126069 -0.08423274 -0.07464009  0.12547043
 -0.05319143 -0.024942    0.00445242 -0.00070851 -0.04380871  0.00809219
  0.02289556  0.13103996]
The puppets look really cheesy , not in a good way like in the Puppet Master 80's flicks.  
[ 0.02925939 -0.03758173 -0.07188705 -0.08595993  0.10257377 -0.01438705
 -0.02089073 -0.09525668 -0.12842315  0.14073167 -0.0426942   0.11572721
 -0.05696136  0.14355302  0.10939836  0.05975785  0.11908917  0.11930049
 -0.04452965 -0.09413064  0.00452817  0.07437155  0.01358453  0.05227087
  0.07

[ 0.01790286  0.09714162 -0.11204261  0.07867005  0.10002156 -0.17614184
 -0.14285332  0.11055003 -0.16997001  0.04103701 -0.08668169 -0.0465235
 -0.04381926 -0.09828966  0.06317837  0.03178358  0.09131284 -0.07643607
 -0.0979557   0.01000519 -0.09479701  0.02061233  0.02509884  0.09821242
  0.03026167 -0.3065788  -0.17003135  0.11327472  0.10395184 -0.13108292
  0.62410409  0.15482335 -0.17641295  0.06315035  0.10186698 -0.00995202
  0.05987057  0.08735378  0.00228797 -0.2095644   0.05591234 -0.02163625
 -0.09590574 -0.1580951   0.05006053  0.11721636 -0.01512495  0.00529869
  0.24746324  0.05550863]
Great audio!.
[-0.03554726  0.14656046  0.01619973  0.05455339 -0.01435565 -0.06108216
 -0.08757152 -0.07779923  0.0499295   0.19251262  0.04094133  0.1507445
  0.01661098 -0.012359   -0.01346866 -0.02064208 -0.01760275  0.10499427
 -0.09036803 -0.08332311  0.09510636  0.07954234  0.03018108 -0.00525567
  0.17745677 -0.21826971 -0.25720855 -0.07247323  0.05061611 -0.0135558
  0.71105963 -

We loved the biscuits!!!
[-0.05090892  0.05065711 -0.18695961  0.00941754  0.17091236  0.16281719
 -0.0358066   0.01511848  0.0479711   0.08991933 -0.02733838  0.17042525
 -0.00166224  0.05734488  0.025418   -0.04776759  0.03668051  0.20605062
 -0.03082455 -0.09619526  0.09973867  0.1487129   0.16999042  0.05524713
  0.14438666 -0.15966746 -0.36832153  0.01524639  0.21536358 -0.13198328
  0.44402648  0.03685279 -0.0739285   0.37025347 -0.02445478  0.02433506
 -0.07023249  0.01169487  0.06262993 -0.10440106  0.09851689 -0.04805917
 -0.1457448  -0.06628466  0.10524553  0.10080784 -0.02066052 -0.25169073
 -0.07366568  0.04328187]
Poor quality.
[-0.1315337  -0.01511493 -0.06584145 -0.07731705  0.02303132 -0.02351349
 -0.07816388 -0.079929    0.14844434  0.11983182  0.07718332  0.01177431
  0.08417807 -0.13710031  0.01484389 -0.01578348  0.02106825  0.01163809
  0.13007745 -0.17270109 -0.0460875   0.07528814 -0.0220967  -0.05549765
  0.05865642 -0.18738358 -0.11116266 -0.01625549  0.1122532

[-0.01478445  0.20042519  0.00794558 -0.01971955  0.10516448  0.067416
 -0.04457193 -0.10206447 -0.08975661 -0.00897633 -0.02624419  0.10572116
 -0.06264014  0.00119596  0.19718129  0.02422472 -0.00669445  0.00233142
 -0.10990946 -0.02683578 -0.02745946  0.03259507  0.13972095  0.10252077
  0.07781051 -0.34436033 -0.1701854  -0.01278865  0.03280743 -0.10294748
  0.75948302  0.02703484 -0.04521356 -0.00377991  0.03417779 -0.00802519
  0.06680562 -0.05545957  0.06632721 -0.1011233   0.02354126  0.07763999
  0.00707991 -0.08289773  0.02501962  0.04703904  0.06808733 -0.12062832
  0.06219178  0.12336325]
If you don't find it, too bad, as again the unit comes with one earpad only.I would not recommend this to anyone.
[ 0.11559249 -0.04954756  0.03472802 -0.0609862   0.11213773  0.00822765
 -0.0682856   0.0394809  -0.01353656  0.00303141  0.00511076  0.10277086
 -0.07758703 -0.00832456  0.1572669   0.11873019  0.06168944 -0.0118655
 -0.00315493 -0.15968792 -0.0026347   0.03033506  0.06439653

I gave it a 10  
[-0.01623969  0.09759514  0.12403293  0.00148446  0.14488889  0.03669794
 -0.18293731  0.03278792 -0.09879705 -0.01640952 -0.05427125 -0.07528903
 -0.06218907 -0.05629479  0.18619156 -0.13087742 -0.08769297 -0.13869384
 -0.24673277 -0.00724254  0.04389546  0.02984596  0.14715557 -0.0701576
 -0.03232352 -0.27258928  0.0809366  -0.09334411 -0.035519   -0.07942598
  0.7438213   0.08632167 -0.03916274  0.05128136  0.09331761 -0.05114146
  0.10850563  0.06573026  0.07205196 -0.07314104  0.03422952  0.01947132
  0.04086477 -0.06838643 -0.09059114  0.04099164 -0.06078896  0.0070017
  0.01977508  0.01644674]
So there is no way for me to plug it in here in the US unless I go by a converter.
[  4.93809716e-02   1.08883509e-02   2.19984879e-01   1.56754810e-02
  -4.74012714e-04  -3.19579372e-02   1.81792241e-02   1.20400823e-02
  -3.29726829e-02   6.91186268e-02  -1.96511332e-02   8.45487317e-02
  -7.50421876e-02   4.41552005e-02   1.02248327e-01   1.97728628e-01
   7.61693819e-0

I don't know what happened in Season Five, what a mess.  
[ 0.04150782  0.00430474 -0.00197203 -0.01646584  0.07853962 -0.01766931
 -0.13935854  0.07716097 -0.09791109 -0.04083538 -0.05854849  0.07306747
 -0.18394441  0.01918236  0.27457754  0.05638738  0.07963689 -0.00188432
 -0.09402967 -0.09608157 -0.10053325  0.01853581  0.13350802  0.03256356
  0.06559231 -0.32335339 -0.12716647  0.01979952  0.17513041 -0.10320156
  0.70760223  0.14783023 -0.05759428 -0.0603806   0.04291028  0.04643564
  0.05919039  0.0434012   0.07200034 -0.07754178 -0.19098608 -0.02723874
 -0.00470635  0.09431877 -0.04393464  0.06508471 -0.01662495  0.03951358
  0.00308683  0.09265326]
Battery lasts only a few hours.
[ 0.08628628  0.03044008  0.13202729  0.06543113 -0.00273303  0.00548954
 -0.01119334  0.11076584  0.02648373  0.00229124 -0.11548199 -0.16224076
 -0.09472427 -0.01264941  0.10462978 -0.05259211 -0.19795014  0.08721747
 -0.26496836 -0.08177959  0.153889    0.03867542  0.0529285  -0.06228433
 -0.0882

[ 0.05306345  0.04322199 -0.05913643 -0.05686041  0.08530762 -0.01795831
 -0.166788   -0.08832365 -0.04268401  0.08794809 -0.01434874  0.08767111
 -0.09027084  0.01321824  0.18488288  0.02877239  0.10839671  0.04214882
 -0.01560062 -0.10680813  0.04933472  0.07473344  0.0132353  -0.00605164
  0.10086557 -0.35188746 -0.20802347 -0.03984276  0.04319394 -0.10531281
  0.75433697  0.04468749 -0.00548136 -0.17740107 -0.01528098  0.02643687
  0.01365661  0.13073731 -0.03035856 -0.06208922 -0.06819127  0.04057636
  0.02182912  0.00658669 -0.01425194  0.01292163 -0.05749663 -0.00286732
 -0.05974083  0.14064147]
It didn't charge for me.
[ 0.08360323 -0.04448999  0.04245871 -0.00850273  0.1419165   0.06556055
  0.03047382  0.02487838 -0.02744264 -0.0198453   0.0062773   0.09798204
 -0.12215652 -0.10250015  0.16336512  0.05145509  0.03834497 -0.01476019
 -0.02577715 -0.13395277  0.04655009  0.01018146  0.06256365 -0.0074049
 -0.07097037 -0.4783862   0.02630953 -0.03694379  0.07418342 -0.03135011
 

[  5.09417738e-02   1.60312572e-01  -1.61859581e-01  -1.29187258e-01
   5.60243094e-02  -1.80536652e-03  -1.21047419e-01   4.66822176e-03
  -3.10992886e-02   1.08869040e-02   1.31919366e-03   1.46136365e-01
  -6.43258575e-02  -5.46641132e-02   4.33111193e-02  -6.81823965e-02
   1.43202383e-01   7.82008209e-02  -6.03950536e-02  -8.85455919e-02
   4.68809544e-02   4.72306264e-02  -6.71583792e-02  -7.18945898e-02
   1.97210665e-01  -3.25122870e-01  -2.38526990e-01  -1.11977701e-01
   5.87491995e-02  -2.28965661e-03   7.34332307e-01   5.43116170e-02
  -1.10131541e-02  -1.23037925e-01  -2.55140337e-02   8.82297051e-02
  -4.11120448e-02   1.26140310e-01   8.86993514e-03   3.54789038e-02
  -3.25509884e-02  -6.09808041e-04  -2.93157632e-02  -7.53212299e-03
  -2.97993211e-02   5.08682412e-02  -2.53597512e-02  -4.70418265e-02
  -1.09320905e-01   1.07066811e-02]
Highly unprofessional and rude to a loyal patron!
[ 0.07712091 -0.09592272 -0.1985376  -0.13620556  0.14365241 -0.07920765
  0.04886174 

Only Pros : Large seating area/ Nice bar area/ Great simple drink menu/ The BEST brick oven pizza with homemade dough!
[-0.03336827  0.07159932 -0.06253589 -0.0736521   0.15156353  0.04374828
 -0.08286855 -0.03048877 -0.08604705  0.01336382 -0.11013118 -0.05508551
  0.05365964  0.10415203  0.01264164  0.0169604   0.0535149   0.08476255
 -0.06666072 -0.22649003  0.08576278 -0.02524454  0.05166077  0.12026837
 -0.04603372 -0.26337539 -0.16255915  0.13367364  0.21045696 -0.06723428
  0.70810208 -0.01150742 -0.14131501  0.15677193  0.03602029  0.07757883
  0.02665315  0.17175645  0.09299104 -0.03185461  0.12491918  0.0643815
 -0.08798996  0.0618578   0.04040671  0.13831225 -0.01111113 -0.11188088
  0.03134108  0.01823461]
Food was so gooodd.
[ -3.26042807e-02  -1.07605691e-01  -3.54531588e-02   1.15318341e-01
  -2.01761520e-03  -1.88657225e-01  -2.01954711e-01  -1.78614045e-01
   1.42375881e-01   2.54359646e-02   2.18488815e-02   7.46474089e-02
   2.02496188e-01  -1.46209758e-01  -9.428235

[ 0.03709818 -0.01789818  0.01749346 -0.05211658  0.09711041 -0.13322791
 -0.0458604   0.15692978 -0.03533466  0.20458277  0.02169464  0.05503435
 -0.06943909 -0.10618515  0.14861918  0.08213877  0.01416694 -0.01530477
  0.03284791 -0.1857758  -0.06524873  0.09431537  0.00452934  0.04401433
  0.2695875  -0.2108727  -0.31669989 -0.13093432  0.1501986   0.00165832
  0.48047043  0.35359206  0.09330431 -0.17910891 -0.06402698 -0.08314399
 -0.09941044  0.04530358 -0.03532078 -0.1834936  -0.0943329  -0.09641202
 -0.04481164  0.14583291  0.0575211   0.05530318  0.04478636  0.10654165
 -0.00976291  0.16041298]
I really liked that.  
[ 0.03662008  0.00441524 -0.03128084 -0.02856109  0.09316767  0.00395371
 -0.11099303 -0.05714852 -0.0763446   0.10801388 -0.05150616  0.11724448
 -0.12925021  0.04075637  0.13274932  0.08605951  0.1316668   0.0818275
  0.00650371 -0.18282335 -0.09386987  0.11453572  0.05333346  0.06935925
  0.14419051 -0.36574773 -0.2437988   0.08976265  0.18388507 -0.19278487
  0

In [84]:
train_v_ = batch_preprocessor_NN(train_validate)
train_v_.columns = ['X', 'y']
Xtrain = train_v_['X']
ytrain = train_v_['y']

In [85]:
clf = MLPClassifier(solver='adam', alpha=1e-5, hidden_layer_sizes=(100, 50))

In [86]:
nn = clf.fit(np.array(list(Xtrain), dtype=np.float), ytrain)

valid_ = batch_preprocessor_NN(validate)

valid_.columns = ['X', 'y']

Xtest = valid_['X']

ytest = valid_['y']

y_pred = nn.predict(np.array(list(Xtest), dtype=np.float))

print y_pred

print np.dot(y_pred, 1-ytest)

count = np.sum(y_pred!=ytest)

print float(count)/len(y_pred)

[0 1 0 1 1 0 1 1 1 1 0 0 1 0 1 0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 0 0 1 1 0 0 0
 0 0 0 0 0 1 0 0 1 0 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 1 1 1 0 1 0 0 0
 1 0 1 1 0 1 0 1 0 1 1 1 1 1 1 0 0 0 1 0 0 0 1 0 0 1 0 1 1 0 1 0 0 0 0 0 1
 1 1 1 1 1 1 1 0 0 0 1 1 0 0 0 0 1 0 1 0 0 1 1 1 1 1 0 1 0 1 0 1 1 0 1 1 0
 0 0 1 1 0 1 1 0 1 1 1 1 0 0 0 1 0 1 0 0 1 0 1 1 0 1 1 1 0 0 1 1 0 1 1 1 0
 1 0 0 1 1 0 0 0 0 0 1 0 1 1 0 0 1 0 1 1 1 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0 1
 0 1 0 1 1 0 0 1 1 1 0 0 1 1 0 1 1 0 1 0 0 1 0 0 0 1 0 0 1 1 0 1 0 1 0 0 0
 1 1 0 0 1 0 0 0 1 1 1 0 1 1 0 1 0 1 1 1 1 1 1 1 0 1 1 0 1 1 0 0 0 0 1 1 0
 1 1 0 0 1 1 1 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0
 0 0 0 1 1 1 0 0 1 0 0 0 0 0 1 0 1 1 1 1 1 0 1 0 1 0 1 1 1 1 0 0 1 1 1 0 0
 1 0 0 1 1 0 1 1 1 0 1 1 1 1 0 1 0 0 0 0 0 0 0 1 0 0 1 1 0 0 0 0 1 1 0 1 1
 1 0 1 0 0 1 0 0 1 1 0 0 0 1 0 1 0 0 1 1 0 1 0 0 0 1 0 0 0 0 0 1 1 0 1 0 1
 1 0 0 1 1 0 1 1 0 1 0 0 1 1 0 0 0 1 0 1 1 0 1 1 0 0 1 1 1 0 0 0 0 0 0 0 0
 1 0 0 1 0 0 1 0 0 0 0 0 

Interestingly, this is better than CNNs! :P Idk. The model probably overfit?
----

Anyway we dont have much to do here, let's try SVMs
---

In [88]:
train_ = batch_preprocessor_NN(train)
train_.columns = ['X', 'y']
Xtrain = train_['X']
ytrain = train_['y']

In [108]:
clf = svm.SVC(C=5.0)

In [105]:
clf = clf.fit(np.array(list(Xtrain), dtype=np.float), ytrain)

In [106]:
y_pred = clf.predict(np.array(list(Xtest), dtype=np.float))

print np.dot(y_pred, 1-ytest)

count = np.sum(y_pred!=ytest)

print float(count)/len(y_pred)

60
0.24


In [109]:
train_v_ = batch_preprocessor_NN(train_validate)
train_v_.columns = ['X', 'y']
Xtrain = train_v_['X']
ytrain = train_v_['y']

In [110]:
clf = clf.fit(np.array(list(Xtrain), dtype=np.float), ytrain)

In [111]:
y_pred = clf.predict(np.array(list(Xtest), dtype=np.float))

print np.dot(y_pred, 1-ytest)

count = np.sum(y_pred!=ytest)

print float(count)/len(y_pred)

57
0.238333333333


So the current rankings are:
-----
    - Neural Networks: 5% Testing Error, ~25% Training Error
    - SVMs: 23% Testing Error, 24% Training Error
    - Lexicon based methods: 20% Testing Error
    
TODO:
-----
    - K-Cross Validation
    - Plot the errors over various parameters
    - ROC?
    - Confusion Matrix